### PR TITLE
Fix dayProps for custom dayComponent

### PR DESCRIPTION
**Pull request by nnnnoel**

### CODE REVIEW

**Comment by quizzy** [comment]
This works for me 🎉

**Comment by nnnnoel** [comment]
Before Merge, we can fix temporally.

```javascript

const CustomDayComponent = () => {
    ...
};

CustomDayComponent.propTypes = Day.propTypes;
```

**Review by mynormeza — APPROVED** [review]

**Comment by jp928** [comment]
Is this merged?

**Comment by mynormeza** [comment]
> Is this merged?

No, waiting for approval of maintainers. You can fix while waiting for merge with the snippet in the comment above, you need to import the Day component from the calendars dir of the lib, but you can't use 'shouldComponentUpdate' or equivalent with the temporary fix

**Comment by sn4f** [comment]
Thanks! Need this too @nnnnoel

**Review by debold-tech — APPROVED** [review]

**Review by sn4f — APPROVED** [review]

**Comment by joncardasis** [comment]
We've been unable to upgrade until we have the `onPress` accessible again. Hoping this can be merged soon 🤞

**Comment by hengkx** [comment]
When can I merge? Thank you very much.

**Comment by Mansouri147** [comment]
Hey guys, in what path should we do that later fix 
```
const CustomDayComponent = () => {
    ...
};

CustomDayComponent.propTypes = Day.propTypes;
```

**Comment by Mansouri147** [comment]
> Before Merge, we can fix temporally.
> 
> ```js
> const CustomDayComponent = () => {
>     ...
> };
> 
> CustomDayComponent.propTypes = Day.propTypes;
> ```
Hey Guys where or in what path should we implement that fix

**Review by tututututututu — COMMENTED** [review]
passed

**Review by tututututututu — APPROVED** [review]
passed

**Review by enagorny — APPROVED** [review]

**Review by Fgl — APPROVED** [review]

**Review by rberruezo — APPROVED** [review]

**Review by kait-IssaAnagreh — APPROVED** [review]

**Comment by nnnnoel** [comment]
Can i hear to reason that why don't merge this PR?

**Review by taynancid — APPROVED** [review]

**Comment by CHarnel** [comment]
Waiting for this to be merged too 🙏

**Comment by Inbal-Tish** [comment]
@nnnnoel Please provide a description for what is the issue you're fixing and what is your solution.

File: src/calendar/day/index.js
Comment (Inbal-Tish) [inline src/calendar/day/index.js:85]: You are in the Day component which means I can use this.props and not extractOwnProps function. This is meant to extract the props of the relevant component, whether it is BasicDay, PeriodDay, or your custom dayComponent, from this.props and pass only them (without Day extra props).

**Comment by Jardel-coder** [comment]
I am using the latest version and the problem persists...

**Comment by Inbal-Tish** [comment]
Can someone describe the issue please?

**Comment by CHarnel** [comment]
Like @JardaCoder said, many documented props that custom day component receives are missing.
Here are the actual props:
```
{
  "accessibilityLabel": "today Wednesday 21 April 2021 ",
  "children": 21,
  "date": {
    "dateString": "2021-04-21",
    "day": 21,
    "month": 4,
    "timestamp": 1618963200000,
    "year": 2021
  },
  "testID": "native.calendar.SELECT_DATE_SLOT-2021-04-21"
}
```

Props like `state` and `marking` are missing, along with all the rest from `DayComponentProps`

**Comment by Inbal-Tish** [comment]
@nnnnoel I see the issue. Still, you need to pass (and spread) `this.props` instead of `extractComponentProps` that has no meaning now...

**Comment by Arise-cn** [comment]
@Inbal-Tish Excuse me, is it solved now？

**Comment by Inbal-Tish** [comment]
> @Inbal-Tish Excuse me, is it solved now？

Once the PR will be fixed I'll be able to merge it and it should solve the issue.

File: src/calendar/day/index.js
Comment (tututututututu) [reply src/calendar/day/index.js:85]: nice

**Review by tututututututu — APPROVED** [review]

**Review by taynancid — APPROVED** [review]

**Comment by devAxick** [comment]
I have installed 1.829.0 version, and it helps me.

**Comment by bobdepreziz** [comment]
When is this fixed expected to be released? This is really blocking for our project. Thanks.

File: src/calendar/day/index.js
Comment (Sunhat) [reply src/calendar/day/index.js:85]: @Inbal-Tish 
Is this the correct answer?
```
const dayProps = this.props
```

**Comment by Sunhat** [comment]
Please @nnnnoel fix as soon as you can, thank you!

File: src/calendar/day/index.js
Comment (Inbal-Tish) [reply src/calendar/day/index.js:85]: Yes but you don't need the 'dayProps` anymore, you can just spread `this.props`

**Comment by Inbal-Tish** [comment]
Fix for this issue will be published on the next release

**Comment by ZeeshanAhmadKhalil** [comment]
> I have installed 1.829.0 version, and it helps me.

Installing that version was giving me `lodash` error

**Comment by Inbal-Tish** [comment]
Find a fix on version 1.1263.0

**Comment by Sunhat** [comment]
Thank you! I was about to make another PR myself!

**Pull request closed without merging**
